R/sql_methods.R

```diff
@@ -54,7 +55,14 @@ read_sql <- function(sql, dbname = NULL, schemaname = NULL, parse_dates = NULL){
  # Unlisting list column types. Also substituting NULL values to NA to remain consistent.
  for (i in seq(1,ncol(result))){
    if (typeof(result[,i]) == "list"){
+     if (class(result[,i][[1]])[1] == "datetime.date"){
+       
+       # returning date format as string if parse_dates is not specified.
+       result[,i] <- unlist(lapply(result[,i], function(x) {if (is.null(x)){NA} else {x$isoformat()}}))
```

Comment:
@pibence why not convert to R date type (as.Date) if python type is datetime.date?

@pibence can we also handle datetime.datetime class - like below:

as.POSIXct(x$timestamp(), origin="1970-01-01")

Reply:
@andrewsali I managed to convert it to POSIXct, which gave me the following idea: leave out the parse_dates option and automatically format every date column to POSIXct if they are stored as dates. This would make the usage easier I think.